tests/js/sentry-test/reactTestingLibrary.tsx

```diff
@@ -36,6 +36,21 @@ function createProvider(contextDefs: Record<string, any>) {
   };
 }
 
+export function BaseProviders(props: {
+  children: React.ReactNode;
+  organization?: Organization;
+}): React.ReactElement {
+  return (
+    <CacheProvider value={cache}>
+      <ThemeProvider theme={lightTheme}>
```

Comment:
since the idea is not to hide anything and make our compositional components explicit, despite the verbosity, this option is the one I liked the most.

- We just have to take into account that this list can grow if we add more `context.provider`, or maybe devs will still need to wrap a component with a not common provider. We currently do that, for instance with [MetricsSwitchContext.Provider](https://github.com/getsentry/sentry/blob/b892677a902f4b04fe2fe70acc2893ef4e69621e/tests/js/spec/views/performance/transactionSummary.spec.tsx#L58)

- In the future, we may want to test using the dark mode. So we may want to make the theme a prop too

Reply:
>We just have to take into account that this list can grow if we add more common context.provider and devs will still need to wrap a component with a not common provider. We currently do that, for instance with [MetricsSwitchContext.Provider](https://github.com/getsentry/sentry/blob/b892677a902f4b04fe2fe70acc2893ef4e69621e/tests/js/spec/views/performance/transactionSummary.spec.tsx#L58)

@priscilawebdev I dont think the list should grow past these 2 providers. It's obviously possible, but we'd have to be sure that it's something that makes sense for each component.

> In the future, we may want to test using the dark mode. So we may want to make the theme a prop too

Does dark mode ever differ in functionality?

Reply:
About the dark mode... sometimes we create visual snapshots in our unit/integration tests.

and we can also test styles, for instance [here](https://github.com/getsentry/sentry/blob/38b4e33fc843d9c7fdb2b66c4fa7dd3d5e938afb/tests/js/spec/components/forms/radioGroup.spec.tsx#L35),  although it's not something I personally like to do often as it sounds like "implementation detail" 

But for sure it's something users can see :)

Reply:
For visual snapshots, I dont think it would make sense to run dark mode tests until we figure out how to speed up those tests. It would bring us a lot of pain and very little value to test light/dark mode.